### PR TITLE
Move credential details to vendor configuration

### PR DESCRIPTION
**Pull request by grolu**
**What this PR does / why we need it**:

This is the third preparatory refactor extracted from #2824 and follows the vendor registry extraction in #3075 and typed vendor lookups in #3082.

Credential detail rendering previously used a provider-name switch in `credential/helper.js`. That kept provider-specific display metadata in application code and required extending the switch for every supported vendor.

This change:

- moves read-only credential-detail definitions into each infrastructure and DNS vendor module under `frontend/src/data/vendors`
- replaces the provider switch with a configuration-driven detail resolver
- supports the existing Azure, vSphere, and RFC2136 Secret-key aliases
- safely resolves GCP project IDs from structured service-account data, including the existing Google Cloud DNS fallback
- looks up vendor configuration through the explicit `infra` or `dns` domain
- renames the DNS-only raw-credential composable to `useDnsProviderCredential`

**Related PRs**:

- Original combined refactor: #2824
- First preparatory refactor: #3075
- Second preparatory refactor: #3082

**Which issue(s) this PR fixes**:

None.

**Special notes for your reviewer**:

Please review this as the third preparatory refactor in the sequential split of #2824. The `secret.details` configuration is read-only display metadata; it is not used when creating or updating Secrets.

Validation:

- `make verify`
- focused credential-helper, infrastructure-binding, and DNS-credential composable tests
- provider-by-provider snapshot coverage for credential detail output
- GitGuardian pre-commit scan

**Release note**:

```other developer
NONE
```

## Summary by CodeRabbit

* **New Features**
  * Added vendor-aware credential details for infrastructure and DNS secrets, including project fields derived from service account content.
  * Improved rendering by passing explicit vendor type into the credential details display.
  * Added support for hidden credential values (e.g., API keys and tokens) in vendor definitions.

* **Bug Fixes**
  * Credential details now safely handle missing/unknown provider data, optional secrets, and malformed or unsupported secret-value mappings.

* **Tests**
  * Expanded Jest coverage for DNS/infra credential resolution, hidden details behavior, and invalid configuration edge cases.

### CODE REVIEW

File: frontend/src/data/vendors/dns/google-clouddns.js [hunk 1]
```diff
@@ -3,4 +3,16 @@ export default {
...
   weight: 400,
   icon: 'google-clouddns.svg',
+  secret: {
+    details: [
+      {
+        label: 'Project',
+        valueFrom: {
+          key: ['serviceaccount.json'],
+          parse: 'json',
+          path: ['project_id'],
```
[the remark this answers is not in the document]
Comment (grolu) [reply frontend/src/data/vendors/dns/google-clouddns.js:13]: this is an old fallback that no longer supported by dns controller

**Review by petersutter — COMMENTED** [review]
/lgtm
/approve
